src/EventTrackerKernel.java

```diff
@@ -0,0 +1,70 @@
+import components.standard.Standard;
+
+/**
+ * Event Tracker kernel component with primary methods.
+ */
+public interface EventTrackerKernel extends Standard<EventTracker> {
```

Comment:
You use IDs. I like it. Where are they generated? How would I get one? When events are added, I assume an ID is generated. None of the methods return that value, so I'm not sure how I would know what ID an event has.